ISPy/util/neural_estimation.py

```diff
@@ -0,0 +1,631 @@
+        """[summary]
+
+        Parameters
+        ----------
+        name : str, optional
+            name of the network, by default 'network1'
+        inputdata : ncfile
+            input file in STiC format
+        logtau : list
+            logtau scale used to train the network
+        original_logtau : list
+            Final stratification of the model to do the interpolation
+        nameoutput : str, optional
+            name of the output model, by default 'model_neuralnetwork.nc'
+
+        Example
+        -------
+        >>> dataprediction = 'newprofiles.nc'
+        >>> original_logtau = sp.model(model_train_list[0],0,0,0).ltau[0,0,0,:]
+        >>> myestimator.prediction(name='network1',dataprediction,logtau,original_logtau,"model_output.nc")
+        
+        """
+
+        print('[INFO] Sending the data to the network')
+        o = sp.profile(inputdata)
+        idx = np.where(o.weights[:, 0] < 1.0)[0]
+        stokelist = np.array([np.concatenate([o.dat[0, :, :, idx, 0], 1e0 * o.dat[0, :, :, idx, 1],
+                                              1e0 * o.dat[0, :, :, idx, 2],
+                                              1e0 * o.dat[0, :, :, idx, 3]])])
+        print(stokelist.shape,'...')
+        self.nl = stokelist.shape[1]
+        self.deepl = deep_network(name, logtau, self.nl)
+        prediction = self.deepl.read_and_predict(stokelist)
+        nx, ny, dum = prediction[0, :, :, :].shape
+        prediction = np.reshape(prediction[0, :, :, :], (nx, ny, 6, len(logtau)))
+        noriginaltau = len(original_logtau)
+
+        # Fill the model with the prediction
+        print('[INFO] Writing in STiC format')
+        m = sp.model(nx=nx, ny=ny, nt=1, ndep=noriginaltau)
+        from tqdm import tqdm
+        for ix in tqdm(range(nx)):
+            for iy in range(ny):
+                temp = np.interp(original_logtau, logtau, np.abs(prediction[ix, iy, 0, :]))
+                vlos = np.interp(original_logtau, logtau, prediction[ix, iy, 1, :])
+                vturb = np.interp(original_logtau, logtau, np.abs(prediction[ix, iy, 2, :]))
+                Bln = np.interp(original_logtau, logtau, prediction[ix, iy, 3, :])
+                Bho = np.interp(original_logtau, logtau, np.abs(prediction[ix, iy, 4, :]))
+                Bazi = np.interp(original_logtau, logtau, prediction[ix, iy, 5, :])
+
+                m.ltau[0, iy, ix, :] = original_logtau
+                m.temp[0, iy, ix, :] = temp * 1e3
+                m.vlos[0, iy, ix, :] = vlos * 1e5
+                m.pgas[0, iy, ix, :] = 1.0
+                m.vturb[0, iy, ix, :] = vturb * 1e5
+                m.Bln[0, iy, ix, :] = Bln * 1e3
+                m.Bho[0, iy, ix, :] = Bho * 1e3
+                m.azi[0, iy, ix, :] = Bazi
+
+        # Write the model
+        m.write(nameoutput)
+
+    def prepare_train(self, model_train_list, stokes_train_list, logtau_train_list):
```

Comment:
Maybe change method name? Because `train` (starting at line 527) is said to "prepare and train the neural network" so it would make more sense to call that function `prepare_train`.

Reply:
Fixed